snakemake/jobs.py

```diff
@@ -316,34 +316,30 @@ def reset_params_and_resources(self):
         self._params = None
 
     @property
-    def conda_env_file(self):
-        if self._conda_env_file is None:
-            expanded_env = self.rule.expand_conda_env(self.wildcards_dict)
-            if expanded_env is not None:
-                # Normalize 'file:///my/path.yml' to '/my/path.yml'
-                if is_local_file(expanded_env):
-                    self._conda_env_file = parse_uri(expanded_env).uri_path
-                else:
-                    self._conda_env_file = expanded_env
-        return self._conda_env_file
+    def conda_env_spec(self):
+        if self._conda_env_spec is None:
+            self._conda_env_spec = self.rule.expand_conda_env(self.wildcards_dict)
+        return self._conda_env_spec
 
     @property
     def conda_env(self):
-        if self.conda_env_file:
+        if self.conda_env_spec:
             if self._conda_env is None:
                 self._conda_env = self.dag.conda_envs.get(
-                    (self.conda_env_file, self.container_img_url)
+                    (self.conda_env_spec, self.container_img_url)
                 )
             return self._conda_env
         return None
 
-    @property
-    def conda_env_path(self):
-        return self.conda_env.path if self.conda_env else None
-
     def archive_conda_env(self):
         """Archive a conda environment into a custom local channel."""
-        if self.conda_env_file:
+        if self.conda_env_spec:
+            if self.conda_env.is_named:
+                raise WorkflowError(
```

Comment:
I suppose that the names conda environment could be exported for reproducibility later?

Reply:
Yes, that should be possible (have to store them by name in the archive then, but I'll have a look.

Reply:
Thought about this again. I don't think we can support that, because named environments could be modified in various ways that we cannot capture with the archiving process, at least not at the moment (it just extracts all conda packages). Of course, in theory one could try to come up with another way of archiving such an environment. But it will at least be hard to find a proper solution that is guaranteed to always work with any kind of modification.

Reply:
Thought about this again. I don't think we can support that, because named environments could be modified in various ways that we cannot capture with the archiving process, at least not at the moment (it just extracts all conda packages). Of course, in theory one could try to come up with another way of archiving such an environment. But it will at least be hard to find a proper solution that is guaranteed to always work with any kind of modification.

Reply:
Open to separate PRs here for the future of course.

Reply:
I think you're right in the limitation.  If we run `conda export` it captures only conda packages and other python packages (eg. installed by `pip`), not any other binaries that may be moved into the environment (how insidious).  But that may be good enough?  On the other hand, there are platform specific differences such that what's exported depends on the OS anyhow, so it's not fully reproducible without a container anyhow.